index.bs

```diff
@@ -275,10 +275,10 @@ A [=lock request=] |request| is said to be <dfn>grantable</dfn> if the following
 1. Let |mode| be |request|'s [=lock request/mode=]
 1. If |mode| is "{{exclusive}}", then return true if all of the following conditions are true, and false otherwise:
     * No [=lock-concept|lock=] in |held| has a [=lock-concept/name=] that equals |name|
-    * No [=lock request=] in |queue| earlier than |request| exists.
-1. Otherwise, mode is "{{shared}}"; return true if all of the following conditions are true, and false otherwise:
+    * Either |queue| [=queue/is empty=], or |request| is the first [=queue/item=] in |queue|.
+1. Otherwise, |mode| is "{{shared}}"; return true if all of the following conditions are true, and false otherwise:
     * No [=lock-concept|lock=] in |held| has [=lock-concept/mode=] "{{exclusive}}" and has a [=lock-concept/name=] that equals |name|.
-    * No [=lock request=] in |queue| earlier than |request| exists.
+    * Either |queue| [=queue/is empty=], or |request| is the first [=queue/item=] in |queue|.
```

Comment:
This rule is same for all modes, maybe it can be deduped?

Reply:
IMHO it's more readable as-is, but suggestions welcome.

Reply:
I mean:

1. If |queue| [=queue/is empty|is not empty=] and |request| is not the first [=queue/item=] in |queue|, return false.
1. If |mode| is "{{exclusive}}", then return true if no [=lock-concept|lock=] in |held| has a [=lock-concept/name=] that equals |name|, or otherwise return false.
1. If |mode| is "{{shared}}", then return true if no [=lock-concept|lock=] in |held| has [=lock-concept/mode=] "{{exclusive}}" and has a [=lock-concept/name=] that equals |name|, or otherwise return false.

Reply:
Yeah, that's not bad. The nested ifs aren't that easy to read, though. (i.e. which "if" does the "otherwise" apply to). Maybe a "switch on |mode|"...

Reply:
Reviewing the current text, it simplifies slightly by removing the second "return", so:

> If queue is not empty and request is not the first item in queue, then return false.
> If mode is exclusive, then return true if no lock in held has a name that equals name, or false otherwise.
> If mode is shared, then return true if no lock in held has mode "exclusive" and has a name that equals name, or false otherwise.

Reply:
Looks cool, and I'm fine with doing that in a separate PR.

Reply:
Probably also simplify "a name" to just "name". 

@pwnall - do you still want to do this as a separate PR? If so, can you LGTM/merge this one? I'm happy to update this PR too - might be better since it captures the discussion.